enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -328,9 +420,99 @@ public void obligation( long txId ) throws IOException
         @Override
         public boolean visit( CommittedTransactionRepresentation transaction ) throws IOException
         {
+            // TODO:
+            //      v Change implementation of "freeze active transactions". How to block new transactions from starting
+            //      - Maybe move from time -> tx ids as safe zone (or support both) -- ??
+            //      - Flush buffered ids to disk (if memory intrusive) -- ??
+            //      - Only do assertOpen after property reads if we are slave or wait for transactions to terminate
+            //      - Place all comments describing the problem / solution in a better place
+            //      v Clean up how we handle Transient...Exception, TxTerminatedException, etc. + how we use Status.*
+
+            // PROBLEM
+            // A slave can read inconsistent or corrupted data (mixed state records) because of reuse of property ids.
+            // This happens when a record that has been read gets reused and then read again or possibly reused in
+            // middle of reading a property chain or dynamic record chain.
+            // This is guarded for in single instance with the delaying of id reuse. This does not cover the Slave
+            // case because the transactions that SET, REMOVE and REUSE the record are applied in batch and thus a
+            // slave transaction can see states from all of the transactions that touches the reused record during its
+            // lifetime.
+            //
+            // PROTOTYPED SOLUTION
+            // Master and Slave agree on quarantine snapshotTime.
```

Comment:
Except each one seems to just read their own configurations and just hope that everyone else in the cluster is configured with a similar "safe zone" duration.